build.zig

```diff
@@ -340,12 +346,29 @@ pub fn build(b: *Build) !void {
         .filters = config.filters orelse &.{},
         .use_llvm = config.use_llvm,
     });
-    unit_tests_exe.root_module.addImport("cli", cli_mod);
+    unit_test_exe.root_module.addImport("cli", cli_mod);
     switch (config.ledger_db) {
-        .rocksdb => unit_tests_exe.root_module.addImport("rocksdb", rocksdb_mod),
+        .rocksdb => unit_test_exe.root_module.addImport("rocksdb", rocksdb_mod),
         .hashmap => {},
     }
-    addInstallAndRun(b, test_step, unit_tests_exe, config);
+    addInstallAndRun(b, unit_test_step, unit_test_exe, config);
+
+    const consensus_test_exe = b.addTest(.{
```

Comment:
I don't see why we need a separate executable for this. I would put the tests in `src` and compile them into the same binary as the other tests. I just don't see a benefit of separating it out at this point. It's inconsistent and seems like a source of confusion. I spoke to dadepo about this already but I'd like to hear your thoughts @InKryption.

These tests should run in ci. This approach is excluding them from running in ci. So we'll either need to change the circleci config or combine it with the unit tests.